Fixing a memory leak in yui example

masrud opened the pull request:
Hi,

This pull request is to fix a memory leak in yui example, issue #1248.
The leak is caused because the view objects are not completely destroyed.

Thanks,
Masrud

samccone commented on examples/yui/js/app.js:
```diff
@@ -115,6 +115,9 @@ YUI.add('todo-app', function (Y) {
 			});
 
+			// To delete previously attached views and fix the memory leak.
+			this.get('container').one('#todo-list').destroy({recursivePurge: true});
```
what a great API flag

`recursivePurge`

samccone commented:
hey @masrud this looks correct to me, but would you mind running a before and after snapshot of the DOM nodes?

masrud commented:
Hi Sam,

yes, just calling destroy does not fix the leak and the flag is required.

I attached a figure of the memory use of the page before and after fixing
the leak. Memory is measured using "capture memory" in Timeline tab of the
developer tools.

In each case, I did the following activities on a page:

for three times
   added 8 tasks, removed 8 tasks, performed a garbage collection

The first figure (yuibefore.png) is generated before fixing the leak, as
can be seen, there is an increasing trend.

The second figure (yuiAfterAddingDestroy.png) is generated after making the
changes in the code (calling destroy function). This fix removes one memory
leak.

There is still another memory leak in the page. This memory leak is caused
by lines 76-80 of the "app.js" file. To fix it, temporarily I commented the
lines.
The third figure (yuiAfterfixingAnotherLeak.png) shows the memory use after
this last fix.

Bests,
Masrud


On Fri, Apr 3, 2015 at 9:48 AM, Sam Saccone <notifications@github.com>
wrote:

> hey @masrud <https://github.com/masrud> this looks correct to me, but
> would you mind running a before and after snapshot of the DOM nodes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tastejs/todomvc/pull/1249#issuecomment-89294260>.
>

masrud commented:
Hi Sam,

yes, just calling destroy does not fix the leak and the flag is required.

I attached a figure of the memory use of the page before and after fixing the leak. Memory is measured using "capture memory" in Timeline tab of the developer tools.

In each case, I did the following activities on a page:

for three times
   added 8 tasks, removed 8 tasks, performed a garbage collection

The first figure (yuibefore.png) is generated before fixing the leak, as can be seen, there is an increasing trend.

The second figure (yuiAfterAddingDestroy.png) is generated after making the changes in the code (calling destroy function). This fix removes one memory leak.

There is still another memory leak in the page. This memory leak is caused by lines 76-80 of the "app.js" file. To fix it, temporarily I commented the lines.
The third figure (yuiAfterfixingAnotherLeak.png) shows the memory use after this last fix.

Bests,
Masrud

![yuibefore](https://cloud.githubusercontent.com/assets/3018144/6983922/35fa920e-d9f1-11e4-85d4-e65010ca66b2.png)
![yuiafteraddingdestroy](https://cloud.githubusercontent.com/assets/3018144/6983928/49f425ea-d9f1-11e4-9ed7-8ca69eb04e80.png)
![yuiafterfixinganotherleak](https://cloud.githubusercontent.com/assets/3018144/6983930/51693f90-d9f1-11e4-96a0-fa0e5e565c6d.png)

samccone commented:
:clap: lgtm anyone else want to take a peak?

addyosmani commented:
This change LGTM (also happy to see more use of profiling tools to nail down our memory leaks). Landing!
